.ado/jobs/cli-init.yml

```diff
@@ -250,12 +268,33 @@ jobs:
               - template: ../variables/vs2019.yml
 
             pool:
-              name: ${{ variables['AgentPool.Medium'] }}
+              ${{ if eq(matrix.lowResource, true) }}:
+                name: ${{ variables['AgentPool.Small'] }}
+              ${{ else }}:
+                name: ${{ variables['AgentPool.Medium'] }}
               demands: ${{ variables['AgentImage'] }}
             timeoutInMinutes: 60
             cancelTimeoutInMinutes: 5
 
             steps:
+              - checkout: self # self represents the repo where the initial Pipelines YAML file was found
+                clean: true # whether to fetch clean each time
+                # fetchDepth: 2 # the depth of commits to ask Git to fetch
```

Comment:
nit: if we could update the publish to verdaccio from using beachball to vanilla `npm publish` we could use a shallow clone with fetchdepth of ~10 to reduce the build time even more.

Reply:
Good way to save time, but would be a bit tricky, since we also rely on it for bumping behavior. We could maybe publish more post-bump version information from the Setup phase though.